### PR TITLE
Updates to compile with Linux 5.10

### PR DESCRIPTION
**Pull request by heitbaum**
bcm_sta: patch to compile with Linux 5.10

- Adapted Joan Bruguera’s patch 
- https://gist.github.com/joanbm/5c640ac074d27fd1d82c74a5b67a1290

xf86-video-nvidia: update to 455.45.01 for compatibility with Linux 5.10

xf86-video-nvidia-legacy: update to 390.138 for compatibility with Linux 5.10
- 5.10 patch from https://build.opensuse.org/package/view_file/X11:Drivers:Video/nvidia-gfxG04/kernel-5.10.patch
- whilst all the patches work, the driver is using GPL code and is thus blocked.
- I have removed xf86-video-nvidia-legacy from the build for the present.
```
FATAL: modpost: GPL-incompatible module nvidia-uvm.ko uses GPL-only symbol 'radix_tree_preloads'
```

### CODE REVIEW

**Comment by chewitt** [comment]
RTL drivers are bumped in #4691

File: distributions/LibreELEC/options [hunk 1]
```diff
@@ -214,7 +214,7 @@
 # Xorg Graphic drivers to use (all / i915,i965,r200,r300,r600,nvidia)
 # Space separated list is supported,
 # e.g. GRAPHIC_DRIVERS="i915 i965 r300 r600 radeonsi nvidia"
-  GRAPHIC_DRIVERS="r300 r600 radeonsi iris i915 i965 nvidia nvidia-legacy vmware virtio"
```
Comment (CvH) [inline distributions/LibreELEC/options:217]: looks like you disable nvidia-legacy :)

File: packages/x11/driver/xf86-video-nvidia-legacy/package.mk [hunk 2]
```diff
@@ -3,8 +3,8 @@
 # Copyright (C) 2016-present Team LibreELEC (https://libreelec.tv)
 
 PKG_NAME="xf86-video-nvidia-legacy"
-PKG_VERSION="340.108"
-PKG_SHA256="995d44fef587ff5284497a47a95d71adbee0c13020d615e940ac928f180f5b77"
+PKG_VERSION="390.138"
```
Comment (CvH) [inline packages/x11/driver/xf86-video-nvidia-legacy/package.mk:6]: anyone an idea if nvidia ion still works with that version ?

**Comment by HiassofT** [comment]
I haven't done any runtime tests but if I use the nvidia legaxy 340 patches from ARCH  https://aur.archlinux.org/cgit/aur.git/tree/?h=nvidia-340xx I only need this small additional patch to build the 340.108 nvidia kernel module with kernel 5.10
```
--- orig/NVIDIA-Linux-x86_64-340.108/kernel/nv-drm.c	2020-12-14 19:58:21.951120294 +0100
+++ new/NVIDIA-Linux-x86_64-340.108/kernel/nv-drm.c	2020-12-14 19:55:27.593332671 +0100
@@ -365,7 +365,7 @@
     struct nv_gem_object *nv_obj = container_of(obj, struct nv_gem_object, base);
     int page_count = obj->size >> PAGE_SHIFT;
 
-    return drm_prime_pages_to_sg(nv_obj->pages, page_count);
+    return drm_prime_pages_to_sg(obj->dev, nv_obj->pages, page_count);
 }
 
 static void* nv_gem_prime_vmap(
```

**Comment by HiassofT** [comment]
I successfully tested nvidia 340.108 driver with above patches on my old Debian buster desktop PC with Nvidia 8600GTS card and locally built 5.10 kernel.

Haven't (and can't) check LE, but I'm quite confident it should work, too.

**Review by CvH — APPROVED** [review]

**Pull request merged**